packages/filter-workspace-packages/src/index.ts

```diff
@@ -43,15 +43,16 @@ export async function filterPackages<T> (
     prefix: string
     workspaceDir: string
     testPattern?: string[]
-  }
+  },
+  filterProd: string[] = []
```

Comment:
It might be better to change the filter argument to a map. Something like:

`filter: Map<string, { followProdDepsOnly: boolean }>`

Reply:
Am I correct in assuming the `string` is the filter in this suggestion?

For example these filters
```
filter=project1
filter=project2
filter-prod=project3
```
would result in the following map?
```
[
  [ "project1", { followProdDepsOnly: false} ],
  [ "project2", { followProdDepsOnly: false} ],
  [ "project3", { followProdDepsOnly: true} ],
]
```

If that is the case I feel like I would need to convert to a map and then back to two different arrays.

Reply:
yes, that was my idea.

If it is better, you could use an objects instead, like:

```
[
  { filter: "project1", followProdDepsOnly: false},
  { filter: "project2", followProdDepsOnly: false},
]
```

Reply:
I'm most familiar with objects so I lean towards using objects. 

Would you expect `filterPkgsBySelectorObjects` to also have the same concept applied? ie. Have the `packageSelectors` arg become a map or array of objects?

Reply:
Yes, I believe it would be better. Less code duplication.

Reply:
I made the changes. Let me know if it matches what you were requesting.